contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolEthereumE2E.t.sol

```diff
@@ -0,0 +1,423 @@
+import {IPriceRegistry} from "../../../interfaces/IPriceRegistry.sol";
+import {IRMN} from "../../../interfaces/IRMN.sol";
+import {RateLimiter} from "../../../libraries/RateLimiter.sol";
+import {Client} from "../../../libraries/Client.sol";
+import {Internal} from "../../../libraries/Internal.sol";
+import {MerkleHelper} from "../../helpers/MerkleHelper.sol";
+import {BaseTest} from "../../BaseTest.t.sol";
+import {E2E} from "../End2End.t.sol";
+import {GhoBaseTest} from "./GhoBaseTest.t.sol";
+
+contract GhoTokenPoolEthereumE2E is E2E, GhoBaseTest {
+  using Internal for Internal.EVM2EVMMessage;
+
+  IBurnMintERC20 internal srcGhoToken;
+  IBurnMintERC20 internal dstGhoToken;
+  UpgradeableLockReleaseTokenPool internal srcGhoTokenPool;
+  UpgradeableBurnMintTokenPool internal dstGhoTokenPool;
+
+  function setUp() public virtual override(E2E, BaseTest) {
+    E2E.setUp();
+
+    // Deploy GHO Token on source chain
+    srcGhoToken = IBurnMintERC20(address(new GhoToken(AAVE_DAO)));
+    deal(address(srcGhoToken), OWNER, type(uint128).max);
+    // Add GHO token to source token list
+    s_sourceTokens.push(address(srcGhoToken));
+
+    // Deploy GHO Token on destination chain
+    dstGhoToken = IBurnMintERC20(address(new GhoToken(AAVE_DAO)));
+    deal(address(dstGhoToken), OWNER, type(uint128).max);
+    // Add GHO token to destination token list
+    s_destTokens.push(address(dstGhoToken));
+
+    // Deploy LockReleaseTokenPool for GHO token on source chain
+    srcGhoTokenPool = UpgradeableLockReleaseTokenPool(
+      _deployUpgradeableLockReleaseTokenPool(
+        address(srcGhoToken),
+        address(s_mockRMN),
+        address(s_sourceRouter),
+        AAVE_DAO,
+        INITIAL_BRIDGE_LIMIT,
+        PROXY_ADMIN
+      )
+    );
+
+    // Add GHO UpgradeableTokenPool to source token pool list
+    s_sourcePoolByToken[address(srcGhoToken)] = address(srcGhoTokenPool);
+    s_destTokenBySourceToken[address(srcGhoToken)] = address(dstGhoToken);
+
+    // Deploy BurnMintTokenPool for GHO token on destination chain
+    dstGhoTokenPool = UpgradeableBurnMintTokenPool(
+      _deployUpgradeableBurnMintTokenPool(
+        address(dstGhoToken),
+        address(s_mockRMN),
+        address(s_destRouter),
+        AAVE_DAO,
+        PROXY_ADMIN
+      )
+    );
+
+    // Add GHO UpgradeableTokenPool to destination token pool list
+    s_sourcePoolByToken[address(dstGhoToken)] = address(dstGhoTokenPool);
+    s_destTokenBySourceToken[address(dstGhoToken)] = address(srcGhoToken);
+
+    // Give mint and burn privileges to destination UpgradeableTokenPool (GHO-specific related)
+    vm.stopPrank();
+    vm.startPrank(AAVE_DAO);
+    GhoToken(address(dstGhoToken)).grantRole(GhoToken(address(dstGhoToken)).FACILITATOR_MANAGER_ROLE(), AAVE_DAO);
+    GhoToken(address(dstGhoToken)).addFacilitator(address(dstGhoTokenPool), "UpgradeableTokenPool", type(uint128).max);
+    vm.stopPrank();
+    vm.startPrank(OWNER);
```

Comment:
honestly i wanted to remove `changePrank` everywhere else because `changePrank` has been deprecated by foundry and if you check the logs with `-vvv` foundry suggests to use `startPrank` instead, those are mostly from the previous test suite (for 1.4)

Reply:
Ah ok cool